app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  validates :email, uniqueness: true, presence: true
+  validates :password, presence: true, on: :create
+  validates :password_digest, presence: true
+
+  has_secure_password
+
+  before_create :set_api_key
+
+  def set_api_key
+    self.api_key = SecureRandom.uuid
+  end
+end
```

Comment:
This is the basic setup of a our user whenever they need to register. Remember bcrypt gem and all the above validations along with the has_secure_password